src/ru/nsu/ccfit/zuev/osu/scoring/StatisticV2.java

```diff
@@ -163,17 +168,21 @@ public int getModifiedTotalScore() {
     }
 
     public int getAutoTotalScore() {
-        float mult = 1;
-        for (GameMod m : mod) {
-            if (m.unranked) {
-                continue;
+        // Optimization :P
+        if (Float.isNaN(rankedScoreMultiplier)) {
```

Comment:
This is to avoid recomputing the scoreMultiplier everytime this method is called (which is called multiple times in one tick of the update thread), it must be safe because the multiplier shouldn't change along a play.